src/ps_modules/VstsTaskSdk/FindFunctions.ps1

```diff
@@ -0,0 +1,728 @@
+        Add-Type -LiteralPath $PSScriptRoot\Minimatch.dll
+
+        # Normalize slashes for root dir.
+        $DefaultRoot = ConvertTo-NormalizedSeparators -Path $DefaultRoot
+
+        $results = @{ }
+        $originalMatchOptions = $MatchOptions
+        foreach ($pat in $Pattern) {
+            Write-Verbose "Pattern: '$pat'"
+
+            # Trim and skip empty.
+            $pat = "$pat".Trim()
+            if (!$pat) {
+                Write-Verbose 'Skipping empty pattern.'
+                continue
+            }
+
+            # Clone match options.
+            $MatchOptions = Copy-MatchOptions -Options $originalMatchOptions
+
+            # Skip comments.
+            if (!$MatchOptions.NoComment -and $pat.StartsWith('#')) {
+                Write-Verbose 'Skipping comment.'
+                continue
+            }
+
+            # Set NoComment. Brace expansion could result in a leading '#'.
+            $MatchOptions.NoComment = $true
+
+            # Determine whether pattern is include or exclude.
+            $negateCount = 0
+            if (!$MatchOptions.NoNegate) {
+                while ($negateCount -lt $pat.Length -and $pat[$negateCount] -eq '!') {
+                    $negateCount++
+                }
+
+                $pat = $pat.Substring($negateCount) # trim leading '!'
+                if ($negateCount) {
+                    Write-Verbose "Trimmed leading '!'. Pattern: '$pat'"
+                }
+            }
+
+            $isIncludePattern = $negateCount -eq 0 -or
+                ($negateCount % 2 -eq 0 -and !$MatchOptions.FlipNegate) -or
+                ($negateCount % 2 -eq 1 -and $MatchOptions.FlipNegate)
+
+            # Set NoNegate. Brace expansion could result in a leading '!'.
+            $MatchOptions.NoNegate = $true
+            $MatchOptions.FlipNegate = $false
+
+            # Trim and skip empty.
+            $pat = "$pat".Trim()
+            if (!$pat) {
+                Write-Verbose 'Skipping empty pattern.'
+                continue
+            }
+
+            # Expand braces - required to accurately interpret findPath.
+            $expanded = $null
+            $preExpanded = $pat
+            if ($MatchOptions.NoBrace) {
+                $expanded = @( $pat )
+            } else {
+                # Convert slashes on Windows before calling braceExpand(). Unfortunately this means braces cannot
+                # be escaped on Windows, this limitation is consistent with current limitations of minimatch (3.0.3).
+                Write-Verbose "Expanding braces."
+                $convertedPattern = $pat -replace '\\', '/'
+                $expanded = [Minimatch.Minimatcher]::BraceExpand(
+                    $convertedPattern,
+                    (ConvertTo-MinimatchOptions -Options $MatchOptions))
+            }
+
+            # Set NoBrace.
+            $MatchOptions.NoBrace = $true
+
+            foreach ($pat in $expanded) {
+                if ($pat -ne $preExpanded) {
```

Comment:
Hey @baywert. This is very important info: everything that is under "VstsTaskSdk" folder is NOT under my authority. This is fully developed by parts of the Product Team of Azure DevOps. They maintain the SDK. For development, they say, that I should download the module as is and pack it with every task that uses this.

https://github.com/Microsoft/azure-pipelines-task-lib/blob/master/powershell/Docs/Consuming.md

The question is: Should we include it directly in the repo to make it easier for everybody to get started. Or, should we rather only have it included in the build pipeline itself (e.g. have it downloaded on build time and packaged)

Reply:
thanks for the precision. Do they have a module published on the gallery? maybe we could leverage install module instead of having to maintain our own copy. (this way consumers don't have to do anything)

Reply:
Yes, there is a PS Gallery entry. For the consumer, there wouldn't be anything to do. It will always be packaged with the extension. It is basically the same way you would do with nodejs packages. You define a version and a package and that is being packed with the whole module.
In this case, it is pre-downloaded and included manually.
But yes, this package could be downloaded with the extension on the agent in case it is not existing already.

So for the end product, it does not make any difference currently, since the SDK does not change much.
I would suggest the following. We make this an improvement issue with another release since I would like to get the extension out first. In my opinion, it is neither a blocker nor does the end user has anything to do to use this task.

What do you say?

Reply:
Yeah I think documenting an improvement issue would be a good middle-ground to get this one out of the door faster. Don't hesitate to include all the technical details on how to do it.

Reply:
I resolve this comment and the enhancement if logged with the issue #3